arrow-libs/fx/arrow-fx-coroutines/src/commonMain/kotlin/arrow/fx/coroutines/flow.kt

```diff
@@ -209,16 +211,17 @@ public fun <A> Flow<A>.repeat(): Flow<A> =
 public fun <A> Flow<A>.metered(period: Duration): Flow<A> =
   fixedRate(period).zip(this) { _, a -> a }
 
+@ExperimentalTime
 public fun <A> Flow<A>.metered(period: Long): Flow<A> =
   fixedRate(period).zip(this) { _, a -> a }
 
 @ExperimentalTime
 public fun fixedRate(
-  period: Duration,
+  periodInMillis: Long,
   dampen: Boolean = true,
-  timeStampInMillis: () -> Long = { timeInMillis() }
+  timeStamp: () -> ComparableTimeMark = { TimeSource.Monotonic.markNow() }
```

Comment:
Would there be any benefit to taking `TimeSource` here instead? Or do we need `Monotonic` for `ComparableTimeMark`?

Reply:
I think we need `ComparableTimeMark` because comparison is used in at least one place. Also, the deprecation warnings from `timeInMillis` mentioned that this was the expected substitution, so I just went with it.